mattermost/mattermost-server
# [MM-36280] Add the possibility to delete user's recent search

#### Summary

Currently a user can only get the list of the 5 most recent searches, in this [figma designs](https://www.figma.com/file/SwCd4uBAt0gAWnGpgjfO92/MM-29044-File-Search-Enhancements?node-id=87%3A32384) it's explained that a user can remove a recent search, so this PR adds the possibility to remove a given search query from the user's search history.

##### QA

1. Run the Mattermost server alongside the webapp
2. Create a testing team and user
3. Once the setup process is done, make a search request:
```bash
curl 'http://localhost:8065/api/v4/teams/:team-id/posts/search' \
  -H 'Authorization: bearer <YOUR-USER-TOKEN>' \
  -H 'Content-Type: application/json' \
  --data-raw '{"terms":"test","is_or_search":false,"include_deleted_channels":true,"time_zone_offset":7200,"page":0,"per_page":20}'
```
4. Make sure that you are getting a success response
5. List the recent searches for the current user:
```bash
curl 'http://localhost:8065/api/v4/users/me/recent_searches' \
  -H 'Authorization: bearer <YOUR-USER-TOKEN>' 

```
6. Copy one of the entries in the returned list
7.  And finally, delete that search from the user's history:
```bash
curl -X DELETE 'http://localhost:8065/api/v4/users/me/recent_searches' \
  -H 'Authorization: bearer <YOUR-USER-TOKEN>' \
  -H 'Content-Type: application/json' \
  --data-raw '{"terms":"test","include_deleted_channels":true,"timezone_offset":7200,"modifier":""}'
```
8. Rerun the step `5` and make sure that the search item was removed

#### Ticket Link

GH Issue: #18841, this issue will be closed by another PR once this one is merged on Mattermost-webapp which will implement the [figma designs](https://www.figma.com/file/SwCd4uBAt0gAWnGpgjfO92/MM-29044-File-Search-Enhancements?node-id=87%3A32384).
JIRA ticket: https://mattermost.atlassian.net/browse/MM-36280

#### Release Note

```release-note
DELETE /users/:user_id/recent_searches:
- Request body: SearchParams
```
